crates/comments/src/directive.rs

```diff
@@ -0,0 +1,96 @@
+#[derive(Debug, PartialEq)]
+pub enum Directive {
+    Format(FormatDirective),
+}
+
+#[derive(Debug, PartialEq)]
+pub enum FormatDirective {
+    Skip,
+    SkipFile,
+}
+
+/// Parse a comment directive
+///
+/// These take the form:
+///
+/// ```text
+/// # <category>: <command> <optional-argument>
+/// ```
+///
+/// Such as:
+///
+/// ```text
+/// # fmt: skip
+/// # fmt: skip file
+/// # fmt: tabular
+/// # fmt: align-right
+/// # lint: skip
+/// # lint: skip rule
+/// ```
+///
+/// Note that directives are applied to the node they are attached to.
+pub fn parse_comment_directive(text: &str) -> Option<Directive> {
+    let text = text.strip_prefix('#')?;
```

Comment:
This needs to be `trim_start_matches('#')` for e.g. `## fmt: skip file`.

I've done this in the downstream PR.

Reply:
I actually purposefully did not do that to ensure that something that comes through here must start with `#`. i.e. `trim_start_matches('#')` would be fine with `fmt: skip file` without any `#`